src/cluster/placement/algo/sharded_helper.go

```diff
@@ -404,7 +406,8 @@ func (ph *helper) generatePlacement() placement.Placement {
 		SetIsSharded(true).
 		SetIsMirrored(ph.opts.IsMirrored()).
 		SetCutoverNanos(ph.opts.PlacementCutoverNanosFn()()).
-		SetMaxShardSetID(maxShardSetID)
+		SetMaxShardSetID(maxShardSetID).
+		SetVersion(ph.version)
```

Comment:
Is this needed? I actually prefer to not have version in the placement algorithm, because it's generating a temporary placement between last version and next version, I'd prefer this to be set to a default/meaningless value like `0` to avoid user depend on the version on this temporary placement, I see you are not using this value in the apis as well? If that's the case, I'd prefer to not set this value here.

Reply:
Discussed offline, we'll clean this up separately